rcamp/tests/test_projects_commands.py

```diff
@@ -0,0 +1,251 @@
+
+        expiration_notifier = ExpirationNotifier()
+        actual_allocations = expiration_notifier.get_expired_allocations_with_no_sent_notification()
+        expected_query_set = [alloc_not_notified]
+
+        self.assertEqual(expected_query_set, list(actual_allocations))
+
+    def test_only_return_allocations_with_passed_expiration_date(self):
+
+        sdate_tz = localize_timezone(2016, 02, 02, 'America/Denver')
+        current_date_for_test_tz = localize_timezone(2018, 02, 02, 'America/Denver')
+        with mock.patch('django.utils.timezone.now', return_value=current_date_for_test_tz):
+
+            edate_in_past_tz = localize_timezone(2017, 02, 02, 'America/Denver')
+            alloc_dict_end_date_passed = {
+                'project': self.proj,
+                'allocation_id': 'ucb7_summit2',
+                'amount': '50000',
+                'start_date': sdate_tz,
+                'end_date': edate_in_past_tz,
+                'expiration_notice_sent': False
+            }
+            alloc_end_date_passed = Allocation.objects.create(**alloc_dict_end_date_passed)
+
+            edate_in_future_tz = localize_timezone(2019, 02, 02, 'America/Denver')
+            alloc_dict_end_date_future = {
+                'project': self.proj,
+                'allocation_id': 'ucb6_summit2',
+                'amount': '50000',
+                'start_date': sdate_tz,
+                'end_date': edate_in_future_tz,
+                'expiration_notice_sent': False
+            }
+            alloc_end_date_future = Allocation.objects.create(**alloc_dict_end_date_future)
+
+            expiration_notifier = ExpirationNotifier()
+            actual_allocations = expiration_notifier.get_expired_allocations_with_no_sent_notification()
+            expected_query_set = [alloc_end_date_passed]
+
+            self.assertEqual(expected_query_set, list(actual_allocations))
+
+    def test_send_expiration_notice_for_expired_allocation(self):
+
+        sdate_tz = localize_timezone(2016, 02, 02, 'America/Denver')
+        edate_tz = localize_timezone(2017, 02, 02, 'America/Denver')
+        alloc_dict_not_notified = {
+            'project': self.proj,
+            'allocation_id': 'ucb7_summit2',
+            'amount': '50000',
+            'start_date': sdate_tz,
+            'end_date': edate_tz,
+            'expiration_notice_sent': False
+        }
+        alloc_not_notified = Allocation.objects.create(**alloc_dict_not_notified)
+
+        with mock.patch('mailer.signals.allocation_expired.send') as mock_send:
+            expiration_notifier = ExpirationNotifier()
+            expiration_notifier.send_expiration_notices()
+
+            args, kwargs = mock_send.call_args
+            actual_model = type(kwargs['allocation'])
+            expected_model = type(alloc_not_notified)
+
+            self.assertEqual(mock_send.call_count, 1)
+            self.assertEqual(expected_model, actual_model)
```

Comment:
Here and elsewhere: When checking if an object is an instance of certain class, you should use `isinstance()` in lieu of `type`. It will verify both that the object is of the correct class, and that it is an instance.

As it is written here, this test will still pass if the `allocation` kwarg is `Allocation` or `Allocation()`.